diff --git a/ferminet/networks.py b/ferminet/networks.py
index 315d932..9a27d0d 100644
--- a/ferminet/networks.py
+++ b/ferminet/networks.py
@@ -543,7 +543,9 @@ def construct_symmetric_features(
   # Construct inputs to next layer
   # h.size == 0 corresponds to unoccupied spin channels.
   g_one = [jnp.mean(h, axis=0, keepdims=True) for h in h_ones if h.size > 0]
-  g_one = [jnp.tile(g, [h_one.shape[0], 1]) for g in g_one]
+  g_one = [
+      jnp.broadcast_to(g, (h_one.shape[0], g.shape[-1])) for g in g_one
+  ]
 
   g_two = [jnp.mean(h, axis=0) for h in h_twos if h.size > 0]
 
